Warning  NU1904 cluttering the Build in Visual Studio

warning NU1904: Package 'System.Text.Encodings.Web' 5.0.0 has a known critical severity vulnerability.

I am not sure if there is a solution, but this is also affecting down-stream projects...

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -50,10 +50,13 @@ jobs:
     - uses: actions/checkout@v4
     - uses: actions/setup-dotnet@v4
       name: Install Current .NET SDK
+    # Building the analyzers (the code fix projects in particular) for the linux runtime
+    # results in some super-old System.* packages with vulnerabilities being pulled in.
+    # Our analyzers don't do anything wrt. trimming and AOT so it's safe to disable them.
     - name: Run trimming test
-      run: dotnet publish -c Release Funcky.TrimmingTest -r linux-x64 --self-contained /p:TreatWarningsAsErrors=true
+      run: dotnet publish -c Release Funcky.TrimmingTest -r linux-x64 --self-contained /p:TreatWarningsAsErrors=true /p:FunckyDisableAnalyzers=true
     - name: Run AOT test
-      run: dotnet publish -c Release Funcky.TrimmingTest -r linux-x64 --self-contained /p:TreatWarningsAsErrors=true /p:FunckyTestAot=true
+      run: dotnet publish -c Release Funcky.TrimmingTest -r linux-x64 --self-contained /p:TreatWarningsAsErrors=true /p:FunckyDisableAnalyzers=true /p:FunckyTestAot=true
 
   nupkg:
     name: Generate NuGet Packages

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -8,9 +8,6 @@ on:
 
 env:
   DOTNET_NOLOGO: 1
-  # This is for some reason set to 'all' on CI which means
-  # we get warnings for transitive dependencies (e.g. from Messerli.CodeStyle -> StyleCop)
-  NuGetAuditMode: direct
 
 jobs:
   build:

--- a/Analyzers.props
+++ b/Analyzers.props
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project>
-	<ItemGroup Label="Funcky Analyzers">
+	<ItemGroup Label="Funcky Analyzers" Condition="'$(FunckyDisableAnalyzers)' != 'true'">
 		<ProjectReference Include="..\Funcky.Analyzers\Funcky.Analyzers\Funcky.Analyzers.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false" PrivateAssets="all" />
 		<ProjectReference Include="..\Funcky.Analyzers\Funcky.Analyzers.CodeFixes\Funcky.Analyzers.CodeFixes.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false" PrivateAssets="all" />
 		<ProjectReference Include="..\Funcky.Analyzers\Funcky.BuiltinAnalyzers\Funcky.BuiltinAnalyzers.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false" PrivateAssets="all" />

--- a/Directory.Packages.props
+++ b/Directory.Packages.props
@@ -30,7 +30,7 @@
 		<PackageVersion Include="Verify.SourceGenerators" Version="1.4.0" />
 	</ItemGroup>
 	<ItemGroup Label="Analyzer">
-		<PackageVersion Include="Microsoft.CodeAnalysis" Version="4.0.1" />
+		<PackageVersion Include="Microsoft.CodeAnalysis" Version="4.12.0" />
 		<PackageVersion Include="Microsoft.CodeAnalysis.CSharp" Version="4.0.1" />
 		<PackageVersion Include="Microsoft.CodeAnalysis.Analyzers" Version="3.11.0" /><!-- We explicitly reference these analyzers to get the latest version even though we might not use the latest version of Microsoft.CodeAnalysis everywhere. -->
 		<PackageVersion Include="Microsoft.CodeAnalysis.CSharp.Analyzer.Testing" Version="1.1.2" />

--- a/Directory.Packages.props
+++ b/Directory.Packages.props
@@ -25,13 +25,16 @@
 		<PackageVersion Include="xunit.v3" Version="1.0.1" />
 		<PackageVersion Include="FsCheck.Xunit" Version="3.0.1" />
 		<PackageVersion Include="coverlet.collector" Version="3.1.2" />
-		<PackageVersion Include="Nuget.Packaging" Version="6.12.1" />
 		<PackageVersion Include="Verify.XUnit" Version="16.8.1" />
 		<PackageVersion Include="Verify.SourceGenerators" Version="1.4.0" />
 		<PackageVersion Include="Microsoft.CodeAnalysis" Version="4.12.0" />
 		<PackageVersion Include="Microsoft.CodeAnalysis.CSharp.Analyzer.Testing" Version="1.1.2" />
 		<PackageVersion Include="Microsoft.CodeAnalysis.CSharp.CodeFix.Testing" Version="1.1.2" />
 		<PackageVersion Include="Microsoft.CodeAnalysis.CSharp.CodeRefactoring.Testing" Version="1.1.2" />
+	</ItemGroup>
+	<ItemGroup Label="Transient Test Dependencies">
+		<PackageVersion Include="Nuget.Packaging" Version="6.12.1" />
+		<PackageVersion Include="xunit.extensibility.execution" Version="2.9.3" />
 	</ItemGroup>
 	<ItemGroup Label="Analyzer">
 		<PackageVersion Include="Microsoft.CodeAnalysis.CSharp" Version="4.0.1" />

--- a/Directory.Packages.props
+++ b/Directory.Packages.props
@@ -6,7 +6,7 @@
 	<ItemGroup Label="Runtime Dependencies">
 		<PackageVersion Include="Microsoft.Bcl.HashCode" Version="1.1.1" />
 		<PackageVersion Include="System.Collections.Immutable" Version="1.7.1" />
-		<PackageVersion Include="System.Text.Json" Version="5.0.0" />
+		<PackageVersion Include="System.Text.Json" Version="5.0.2" />
 		<PackageVersion Include="xunit.assert" Version="[2.9.3, 2.10)" />
 		<PackageVersion Include="xunit.extensibility.core" Version="[2.9.3, 2.10)" />
 		<PackageVersion Include="xunit.v3.assert" Version="[1.0.1, 2)" />

--- a/Directory.Packages.props
+++ b/Directory.Packages.props
@@ -19,7 +19,7 @@
 		<PackageVersion Include="Microsoft.CodeAnalysis.PublicApiAnalyzers" Version="3.3.3" />
 	</ItemGroup>
 	<ItemGroup Label="Test Dependencies">
-		<PackageVersion Include="Microsoft.NET.Test.Sdk" Version="17.2.0" />
+		<PackageVersion Include="Microsoft.NET.Test.Sdk" Version="17.12.0" />
 		<PackageVersion Include="xunit" Version="2.9.3" />
 		<PackageVersion Include="xunit.runner.visualstudio" Version="3.0.1" />
 		<PackageVersion Include="xunit.v3" Version="1.0.1" />

--- a/Directory.Packages.props
+++ b/Directory.Packages.props
@@ -28,14 +28,14 @@
 		<PackageVersion Include="Nuget.Packaging" Version="6.12.1" />
 		<PackageVersion Include="Verify.XUnit" Version="16.8.1" />
 		<PackageVersion Include="Verify.SourceGenerators" Version="1.4.0" />
-	</ItemGroup>
-	<ItemGroup Label="Analyzer">
 		<PackageVersion Include="Microsoft.CodeAnalysis" Version="4.12.0" />
-		<PackageVersion Include="Microsoft.CodeAnalysis.CSharp" Version="4.0.1" />
-		<PackageVersion Include="Microsoft.CodeAnalysis.Analyzers" Version="3.11.0" /><!-- We explicitly reference these analyzers to get the latest version even though we might not use the latest version of Microsoft.CodeAnalysis everywhere. -->
 		<PackageVersion Include="Microsoft.CodeAnalysis.CSharp.Analyzer.Testing" Version="1.1.2" />
 		<PackageVersion Include="Microsoft.CodeAnalysis.CSharp.CodeFix.Testing" Version="1.1.2" />
 		<PackageVersion Include="Microsoft.CodeAnalysis.CSharp.CodeRefactoring.Testing" Version="1.1.2" />
+	</ItemGroup>
+	<ItemGroup Label="Analyzer">
+		<PackageVersion Include="Microsoft.CodeAnalysis.CSharp" Version="4.0.1" />
+		<PackageVersion Include="Microsoft.CodeAnalysis.Analyzers" Version="3.11.0" /><!-- We explicitly reference these analyzers to get the latest version even though we might not use the latest version of Microsoft.CodeAnalysis everywhere. -->
 		<PackageVersion Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.0.1" />
 		<PackageVersion Include="Microsoft.VSSDK.BuildTools" Version="17.0.1597"/>
 	</ItemGroup>

--- a/Directory.Packages.props
+++ b/Directory.Packages.props
@@ -25,6 +25,7 @@
 		<PackageVersion Include="xunit.v3" Version="1.0.1" />
 		<PackageVersion Include="FsCheck.Xunit" Version="3.0.1" />
 		<PackageVersion Include="coverlet.collector" Version="3.1.2" />
+		<PackageVersion Include="Nuget.Packaging" Version="6.12.1" />
 		<PackageVersion Include="Verify.XUnit" Version="16.8.1" />
 		<PackageVersion Include="Verify.SourceGenerators" Version="1.4.0" />
 	</ItemGroup>

--- a/Funcky.Analyzers/Funcky.Analyzers.Test/Funcky.Analyzers.Test.csproj
+++ b/Funcky.Analyzers/Funcky.Analyzers.Test/Funcky.Analyzers.Test.csproj
@@ -19,6 +19,8 @@
         <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Analyzer.Testing" />
         <PackageReference Include="Microsoft.CodeAnalysis.CSharp.CodeFix.Testing" />
         <PackageReference Include="Microsoft.CodeAnalysis.CSharp.CodeRefactoring.Testing" />
+        <!-- Microsoft.CodeAnalysis.CSharp.Analyzer.Testing depends on an old version of Nuget.Packaging which pulls in vulnerable packages -->
+        <PackageReference Include="Nuget.Packaging" />
     </ItemGroup>
 
     <ItemGroup>

--- a/Funcky.FsCheck/Funcky.FsCheck.fsproj
+++ b/Funcky.FsCheck/Funcky.FsCheck.fsproj
@@ -9,6 +9,7 @@
     </ItemGroup>
     <ItemGroup>
         <PackageReference Include="FsCheck.Xunit" />
+        <PackageReference Include="xunit.extensibility.execution" />
     </ItemGroup>
     <ItemGroup>
         <Compile Include="FunckyGenerators.fs" />

